### DISCUSSION TITLE
Face Recognition, how do you turn it on?  Not showing up in Explore

### DISCUSSION THREAD

**Discussion by untenops** [immich-app/immich#2671] [Q&A]
Been keeping up with the updates, but I must have missed something.   I do see Recognize faces in the Job menu.  And I have run it.  Seems to count down through my pictures.  But even if I hit the missing button again it just seem to start all over again.  Either way I never see anything about faces or people under Explore, only Places and Things.
   
My Compose and .env are slightly modified as per Issue #776 

Here is a copy of my compose file if I have anything setup wrong

    version: "3.8"

    services:
      immich-server:
        user: "${UID}:${GID}" # we added this line to get a specific user / group id
        image: altran1502/immich-server:release
    #    entrypoint: ["/bin/sh", "./start-server.sh"]
        command: ["start-server.sh"]
        volumes:
          - ${UPLOAD_LOCATION}:/usr/src/app/upload
        env_file:
          - .env
        environment:
          - NODE_ENV=production
       depends_on:
         - redis
         - database
         - typesense
       restart: always

     immich-microservices:
        user: "${UID}:${GID}" # we added this line to get a specific user / group id
        image: altran1502/immich-server:release
    #    entrypoint: ["/bin/sh", "./start-microservices.sh"]
        command: ["start-microservices.sh"]
        volumes:
          - ${UPLOAD_LOCATION}:/usr/src/app/upload
          - ${GEOCODING_DATA}:${REVERSE_GEOCODING_DUMP_DIRECTORY}
        env_file:
          - .env
       environment:
          - NODE_ENV=production
        depends_on:
         - redis
         - database
         - typesense
        restart: always

      immich-machine-learning:
        user: "${UID}:${GID}" # we added this line to get a specific user / group id
        image: altran1502/immich-machine-learning:release
        entrypoint: ["/bin/sh", "./entrypoint.sh"]
        volumes:
          - ${UPLOAD_LOCATION}:/usr/src/app/upload
        env_file:
          - .env
        environment:
          - NODE_ENV=production
        depends_on:
          - database
        restart: always

      immich-web:
        user: "${UID}:${GID}" # we added this line to get a specific user / group id
        image: altran1502/immich-web:release
    #    entrypoint: ["/bin/sh", "./entrypoint.sh"]
        env_file:
          - .env
        restart: always

      redis:
        container_name: immich_redis
        image: redis:6.2
        restart: always

      database:
        container_name: immich_postgres
         image: postgres:14
         env_file:
            - .env
         environment:
           POSTGRES_PASSWORD: ${DB_PASSWORD}
           POSTGRES_USER: ${DB_USERNAME}
           POSTGRES_DB: ${DB_DATABASE_NAME}
           PG_DATA: /var/lib/postgresql/data
        volumes:
          - pgdata:/var/lib/postgresql/data
        restart: always

      immich-proxy:
       container_name: immich_proxy
       image: altran1502/immich-proxy:release
       ports:
         - 2283:8080
       logging:
          driver: none
       depends_on:
          - immich-server
       restart: always

      typesense:
        container_name: immich_typesense
        image: typesense/typesense:0.24.0
        environment:
          - TYPESENSE_API_KEY=${TYPESENSE_API_KEY}
          - TYPESENSE_DATA_DIR=/data
        logging:
          driver: none
        volumes:
          - tsdata:/data
        restart: always

    volumes:
      pgdata:
      tsdata:

**Comment by alextran1502** [comment]
You don't need to turn it on, it should run automatically. Can you provide the log of machine learning container and microservices container?

**Reply by untenops** [reply]
Pulled from Portainer.

The machine learning container:

/bin/sh: 0: cannot open ./entrypoint.sh: No such file
/bin/sh: 0: cannot open ./entrypoint.sh: No such file
/bin/sh: 0: cannot open ./entrypoint.sh: No such file
/bin/sh: 0: cannot open ./entrypoint.sh: No such file
/bin/sh: 0: cannot open ./entrypoint.sh: No such file
/bin/sh: 0: cannot open ./entrypoint.sh: No such file
/bin/sh: 0: cannot open ./entrypoint.sh: No such file
/bin/sh: 0: cannot open ./entrypoint.sh: No such file
/bin/sh: 0: cannot open ./entrypoint.sh: No such file

The microserver:
 a bunch of lines that look like

    [Nest] 1  - 06/06/2023, 1:13:02 AM   ERROR [JobService] Unable to run job handler: Error: getaddrinfo ENOTFOUND immich-machine-learning
    [Nest] 1  - 06/06/2023, 1:13:02 AM   ERROR [JobService] Error: getaddrinfo ENOTFOUND immich-machine-learning
      at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:107:26)

    [Nest] 1  - 06/06/2023, 1:13:02 AM   ERROR [JobService] Object:

    {

      "id": "07714c76-bebf-43a8-90c7-98475336b609"

    }

    [Nest] 1  - 06/06/2023, 1:13:02 AM   ERROR [JobService] Unable to run job handler: Error: getaddrinfo ENOTFOUND immich-machine-learning
    [Nest] 1  - 06/06/2023, 1:13:02 AM   ERROR [JobService] Error: getaddrinfo ENOTFOUND immich-machine-learning

        at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:107:26)

    [Nest] 1  - 06/06/2023, 1:13:02 AM   ERROR [JobService] Object:

    {

      "id": "df497c20-5b43-462e-8b83-ba77020b3292"

    }

**Reply by bobnvic** [reply]
I'm having the same issue with very similar logs (well, identical machine learning logs, my microservices logs also includes a couple of additional errors for output file already exists).

**Reply by jrasm91** [reply]
I believe you can delete the entry point line from the machine learning entry in the docker compose as it's not needed anymore.

**Reply by untenops** [reply]
I just tried that and the machine learning container log was full of errors and restarts. 

    INFO:     Started server process [1]
    INFO:     Waiting for application startup.
    ERROR:    Traceback (most recent call last):
      File "/opt/venv/lib/python3.10/site-packages/starlette/routing.py", line 677, in lifespan
        async with self.lifespan_context(app) as maybe_state:
      File "/opt/venv/lib/python3.10/site-packages/starlette/routing.py", line 566, in __aenter__
        await self._router.startup()
      File "/opt/venv/lib/python3.10/site-packages/starlette/routing.py", line 654, in startup
    await handler()
      File "/usr/src/app/src/main.py", line 56, in startup_event
        get_cached_model(model_name, model_type)
      File "/usr/src/app/src/main.py", line 142, in get_cached_model
        model = _get_model(model, task)
      File "/usr/src/app/src/main.py", line 160, in _get_model
        model = pipeline(model=model, task=task)
      File "/opt/venv/lib/python3.10/site-packages/transformers/pipelines/__init__.py", line 705, in pipeline
        config = AutoConfig.from_pretrained(model, _from_pipeline=task, **hub_kwargs, **model_kwargs)
      File "/opt/venv/lib/python3.10/site-packages/transformers/models/auto/configuration_auto.py", line 928, in from_pretrained
        config_dict, unused_kwargs = PretrainedConfig.get_config_dict(pretrained_model_name_or_path, **kwargs)
      File "/opt/venv/lib/python3.10/site-packages/transformers/configuration_utils.py", line 574, in get_config_dict
        config_dict, kwargs = cls._get_config_dict(pretrained_model_name_or_path, **kwargs)
      File "/opt/venv/lib/python3.10/site-packages/transformers/configuration_utils.py", line 629, in _get_config_dict
        resolved_config_file = cached_file(
      File "/opt/venv/lib/python3.10/site-packages/transformers/utils/hub.py", line 417, in cached_file
        resolved_file = hf_hub_download(
      File "/opt/venv/lib/python3.10/site-packages/huggingface_hub/utils/_validators.py", line 120, in _inner_fn
        return fn(*args, **kwargs)
      File "/opt/venv/lib/python3.10/site-packages/huggingface_hub/file_download.py", line 1159, in hf_hub_download
        os.makedirs(storage_folder, exist_ok=True)
      File "/usr/local/lib/python3.10/os.py", line 215, in makedirs
        makedirs(head, exist_ok=exist_ok)
      File "/usr/local/lib/python3.10/os.py", line 225, in makedirs
        mkdir(name, mode)
    PermissionError: [Errno 13] Permission denied: '/cache'
    ERROR:    Application startup failed. Exiting.
    Matplotlib created a temporary config/cache directory at /tmp/matplotlib-bg_qw27o because the default path (/.config/matplotlib) is not a writable directory; it is highly recommended to set the MPLCONFIGDIR environment variable to a writable directory, in particular to speed up the import of Matplotlib and to better support multiprocessing.

    There was a problem when trying to write in your cache folder (/cache). You should set the environment variable TRANSFORMERS_CACHE to a writable directory.

**Reply by samip5** [reply]
You seem to be missing the /cache mount, please use the latest compose yaml from the 1.60.0 release.

**Comment by ShinasShaji** [comment]
I do not know if this is the issue, but Immich seems to have moved to using `ghcr.io/immich-app` for pulling its images. Perhaps the Maintainers/Contributors can confirm whether it could affect functionality.

Perhaps try with an updated `docker-compose` from the latest release [here ](https://github.com/immich-app/immich/releases/tag/v1.60.0) and see if that helps.

**Comment by untenops** [comment]
Thanks everyone, with the comments I was able to get it working.  I replaced my docker-compose with the current one 1.60 and then made a few changes to it in order to deal with the permission issue seen in #776 

docker-compose.yml

    version: "3.8"

    services:
      immich-server:
        user: "${UID}:${GID}" # we added this line to get a specific user / group id
        container_name: immich_server
        image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
        command: ["start-server.sh"]
        volumes:
          - ${UPLOAD_LOCATION}:/usr/src/app/upload
        env_file:
          - .env
        depends_on:
      - redis
      - database
      - typesense
    restart: always

      immich-microservices:
        user: "${UID}:${GID}" # we added this line to get a specific user / group id
        container_name: immich_microservices
        image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
        command: ["start-microservices.sh"]
        volumes:
          - ${UPLOAD_LOCATION}:/usr/src/app/upload
          - ${GEOCODING_DATA}:${REVERSE_GEOCODING_DUMP_DIRECTORY}  #added for issue 776
        env_file:
          - .env
        depends_on:
          - redis
          - database
          - typesense
        restart: always

      immich-machine-learning:
        user: "${UID}:${GID}" # we added this line to get a specific user / group id
        container_name: immich_machine_learning
        image: ghcr.io/immich-app/immich-machine-learning:${IMMICH_VERSION:-release}
        volumes:
          - ${UPLOAD_LOCATION}:/usr/src/app/upload
          - ${CACHE_DATA}:/cache #MOD TO PUT CACHE IN CUSTOM LOCATION .env  #added for issue 776
        env_file:
          - .env
        restart: always

      immich-web:
        user: "${UID}:${GID}" # we added this line to get a specific user / group id
        container_name: immich_web
        image: ghcr.io/immich-app/immich-web:${IMMICH_VERSION:-release}
        env_file:
          - .env
        restart: always

      typesense:
        container_name: immich_typesense
        image: typesense/typesense:0.24.0
        environment:
          - TYPESENSE_API_KEY=${TYPESENSE_API_KEY}
          - TYPESENSE_DATA_DIR=/data
        logging:
          driver: none
        volumes:
          - tsdata:/data
        restart: always

      redis:
        container_name: immich_redis
        image: redis:6.2
        restart: always
      database:
        container_name: immich_postgres
        image: postgres:14
        env_file:
          - .env
        environment:
          POSTGRES_PASSWORD: ${DB_PASSWORD} 
          POSTGRES_USER: ${DB_USERNAME}
          POSTGRES_DB: ${DB_DATABASE_NAME}
          PG_DATA: /var/lib/postgresql/data
        volumes:
          - pgdata:/var/lib/postgresql/data
        restart: always

      immich-proxy:
        container_name: immich_proxy
        image: ghcr.io/immich-app/immich-proxy:${IMMICH_VERSION:-release}
        environment:
          # Make sure these values get passed through from the env file
          - IMMICH_SERVER_URL
          - IMMICH_WEB_URL
        ports:
          - 2283:8080
        depends_on:
          - immich-server
        restart: always

    volumes:
      pgdata:
      # model-cache:
      tsdata:

And a couple of changes in .env
namely

    # user 
    UID=1000
    GID=1000


    # GEOCODING_DATA=/path/to/an/empty/folder/with/full/permissions/to/immich/user
    GEOCODING_DATA=/????/??????/docker/immich/geo
    REVERSE_GEOCODING_DUMP_DIRECTORY=/usr/src/app/geocoding

    # CACHE LOCATION
    CACHE_DATA=/????/??????/docker/immich/cache
